Google.Api.Gax.Grpc/RpcExceptionExtensions.cs

```diff
@@ -0,0 +1,107 @@
+﻿/*
+ * Copyright 2021 Google LLC
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file or at
+ * https://developers.google.com/open-source/licenses/bsd
+ */
+
+using Google.Protobuf;
+using Google.Protobuf.WellKnownTypes;
+using Google.Rpc;
+using Grpc.Core;
+using System.Linq;
+using Status = Google.Rpc.Status;
+
+namespace Google.Api.Gax.Grpc
+{
+    /// <summary>
+    /// Utility extension methods to make it easier to retrieve extended error information from an <see cref="RpcException"/>.
+    /// </summary>
+    public static class RpcExceptionExtensions
+    {
+        // Visible for testing.
+        internal const string StatusDetailsTrailerName = "grpc-status-details-bin";
+
+        /// <summary>
+        /// Retrieves the <see cref="Status"/> message containing extended error information
+        /// from the trailers in an <see cref="RpcException"/>, if present.
+        /// </summary>
+        /// <param name="ex">The RPC exception to retrieve details from. Must not be null.</param>
+        /// <returns>The <see cref="Status"/> message specified in the exception, or null
+        /// if there is no such information.</returns>
+        public static Rpc.Status GetRpcStatus(this RpcException ex) =>
+            DecodeTrailer(GetTrailer(ex, StatusDetailsTrailerName), Status.Parser);
+
+        /// <summary>
+        /// Retrieves the <see cref="BadRequest"/> message containing extended error information
+        /// from the trailers in an <see cref="RpcException"/>, if present.
+        /// </summary>
+        /// <param name="ex">The RPC exception to retrieve details from. Must not be null.</param>
+        /// <returns>The <see cref="BadRequest"/> message specified in the exception, or null
+        /// if there is no such information.</returns>
+        public static BadRequest GetBadRequest(this RpcException ex) => GetStatusDetail<BadRequest>(ex);
+
+        /// <summary>
+        /// Retrieves the <see cref="ErrorInfo"/> message containing extended error information
+        /// from the trailers in an <see cref="RpcException"/>, if present.
+        /// </summary>
+        /// <param name="ex">The RPC exception to retrieve details from. Must not be null.</param>
+        /// <returns>The <see cref="ErrorInfo"/> message specified in the exception, or null
+        /// if there is no such information.</returns>
+        public static ErrorInfo GetErrorInfo(this RpcException ex) => GetStatusDetail<ErrorInfo>(ex);
```

Comment:
I would personally prefer not to do that to start with. It's easy to add them later on, but we couldn't take them away.

@amanda-tarafa do you agree, or do you think it's worth putting them on immediately?

(Note that `GetMetadata` would have to return an `IDictionary<string, string>` or similar, not just a string.)

Reply:
> (Note that GetMetadata would have to return an IDictionary<string, string> or similar, not just a string.)
Agree, the metadata is map<string, string> error_details.proto.

Node.js, Go and Ruby promote the values now.
It's up to @jskeet and @amanda-tarafa decide, thanks to [help](https://cloud.google.com/dotnet/docs/reference/help/troubleshooting) document, it could surface ErrorInfo to user.